MSC3419: Allow guests to send more event types (#3419)
matrix-org/matrix-doc: proposals/3419-guest-state-events.md

# Guest State Events

## Problem

Currently guest users are not allowed to create arbitrary state events.  This is a problem for MSC3401, where it would be useful for guests to be able to create `m.call.member` events in order to participate in a call.

## Proposal

Let guests send arbitrary state events much like a normal user. Servers may rate limit state events from guests much more aggressively to mitigate abuse.

## Security considerations

> The only reason that guests are denied from performing certain operations is to avoid malicious unauthorised users consuming resources and causing a DoS.  In this instance, sending state events can be quite resource intensive, so if we didn't have a good use case that needed them it'd be right to veto them.

Reviewer: It might be worth mentioning that [by default](https://matrix.org/docs/spec/client_server/latest#m-room-power-levels) guests still wouldn't be able to send state since `state_default` = 50 and `users_default` = 0.

So for the given use-case you'd want to specify `events: {"m.call.member": 0}`, which would further limit the harm that guests can do.

Pretty much, this isn't allowing guests to send arbitrary state, by default, so for most rooms this will be a no-op.

Author: done

# Guest State Events

## Problem

> Currently guest users are not allowed to create arbitrary state events.  This is a problem for MSC3401, where it would be useful for guests to be able to create `m.call.member` events in order to participate in a call.

Reviewer 2: ```suggestion
Currently guest users are not allowed to create arbitrary state events.  This is a problem for [MSC3401](https://github.com/matrix-org/matrix-doc/pull/3401), where it would be useful for guests to be able to create `m.call.member` events in order to participate in a call.
```

## Proposal

Let guests send arbitrary state events much like a normal user. Servers may rate limit state events from guests much more aggressively to mitigate abuse.

> We also relax the arbitrary requirement that guests are only allowed to send `m.room.message` events, but any kind of event allowed by the room's power level configuration as if they were a normal user.

Reviewer 2: ```suggestion
We also relax the existing requirement that guests are only allowed to send `m.room.message` events, instead allowing them to send any kind of event allowed by the room's power level configuration as if they were a normal user.
```
